### PR TITLE
fix(timepicker): fix input data validation for timepicker #3549 #3288

### LINKED ISSUE #3549: feat(timepicker): Add validation for timepicker
Add validation for timepicker (according to #3265)

### PR DESCRIPTION
**Pull request by IraErshova**
Fix input validation for timepicker based on ui-bootstrap validation.
Fix min-max validation issue.

close #3549 #3288

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.

### CODE REVIEW

File: src/timepicker/reducer/timepicker.reducer.ts [hunk 1]
```diff
@@ -50,6 +52,12 @@ export function timepickerReducer(state = initialState, action: Action) {
 
       const _newTime = changeTime(state.value, { hour: action.payload.step });
 
+      if (state.config.max || state.config.min) {
```
Comment (IlyaSurmay) [inline src/timepicker/reducer/timepicker.reducer.ts]: looks like these 2 `if` could be one

File: src/timepicker/timepicker.component.ts [hunk 2]
```diff
@@ -175,33 +189,90 @@ export class TimepickerComponent
...
     this._store.dispatch(
       this._timepickerActions.changeSeconds({ step, source })
     );
   }
 
   updateHours(hours: string): void {
+    this.resetValidation();
     this.hours = hours;
+
+    const isValid = isHourInputValid(this.hours, this.isPM())
```
Comment (IlyaSurmay) [inline src/timepicker/timepicker.component.ts]: isInputLimitValid is called 3 times with the same parameters in this component, maybe it's better to make a function from it.

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

File: src/timepicker/reducer/timepicker.actions.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { Action } from '../../mini-ngrx/index';
+import { Action } from '../../mini-ngrx';
```
Comment (valorkin) [inline src/timepicker/reducer/timepicker.actions.ts]: `index` is must for system.js

Comment (valorkin) [reply src/timepicker/reducer/timepicker.actions.ts]: and in other places of PR too

**Review by valorkin — CHANGES_REQUESTED** [review]
please fix imports

**Comment by YevheniiaMazur** [comment]
Needs to fix conflicting files

**Review by valorkin — APPROVED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**